internal/manager/controllerdef.go

```diff
@@ -250,11 +260,17 @@ func setupControllers(
 				// StatusQueue:       kubernetesStatusQueue,
 			},
 		},
+		// KongUpstreamPolicy controller without HTTPRoute.
 		{
-			Enabled: c.KongUpstreamPolicyEnabled,
+			Enabled:    c.KongUpstreamPolicyEnabled,
+			Controller: kongUpstreamPolicyReconciler,
+		},
+		// HTTPRoute controller for updating KongUpstreamPolicy's ancestor status with services used as backends of HTTPRoutes.
+		{
+			Enabled: c.KongUpstreamPolicyEnabled && c.GatewayAPIHTTPRouteController,
```

Comment:
As I understand, if we have Gateway API CRDs installed, this will run two controllers (`kongUpstreamPolicyReconciler` and the decorated one). I think we should make this mutually exclusive - only one of them should be running, otherwise we'll be processing two requests per KUP update.

Reply:
I added one more controller because I want to start watching `HTTPRoute` when the CRD is installed. If we think it is OK to require restart of KIC to enable watching of `HTTPRoute`, it should be much easier to implement and fewer chances for bugs.

Reply:
From the test results, I found there is no simple way to enable it to watch `HTTPRoute` dynamically. So I removed the extra controller and added a `CRDExist` check in `KongUpstreamPolicy` controller.